nilearn/_utils/tests/test_common.py

```diff
@@ -0,0 +1,51 @@
+import numpy as np
+import pytest
+
+from nilearn._utils import all_classes, all_functions, all_modules
+
+
+@pytest.mark.parametrize("func", [all_modules, all_functions, all_classes])
+def test_all_modules_error(func):
+    with pytest.raises(
+        ValueError,
+        match=(
+            "`modules_to_ignore` and "
+            "`modules_to_consider` cannot "
+            "be both specified."
+        ),
+    ):
+        func(modules_to_ignore=["foo"], modules_to_consider=["bar"])
+
+
+@pytest.mark.parametrize("func", [all_functions, all_classes])
+def test_private_vs_public(func):
+    public_only = set(func(return_private=False))
+    private_and_public = set(func(return_private=True))
+    assert public_only.issubset(private_and_public)
+    assert np.all(
+        [elt[0].startswith("_") for elt in private_and_public - public_only]
+    )
+
+
+def test_all_functions():
```

Comment:
that test would fail every time a public function is added to the API

Reply:
Yes, put this explicitly in the test docstring.

Reply:
actually: do you think I should just modify this test to implement what I mention in the comment [below](https://github.com/nilearn/nilearn/pull/2991#issuecomment-1673725380)

Reply:
@Remi-Gau I think it's a good idea. But feel free to do it in another PR